src/analysis/normed_space/hahn_banach.lean

```diff
@@ -21,6 +21,32 @@ Prove more corollaries
 
 -/
 
+universes u v
+
+/-- Hahn-Banach theorem for continuous linear functions. -/
+class has_exists_extension_norm_eq (𝕜 : Type u)
+  [nondiscrete_normed_field 𝕜] [normed_algebra ℝ 𝕜] : Prop :=
+(exists_extension_norm_eq : ∀
+  (E : Type v)
+  [normed_group E] [normed_space 𝕜 E]
+  (p : subspace 𝕜 E)
+  (f : p →L[𝕜] 𝕜),
+  ∃ g : E →L[𝕜] 𝕜, (∀ x : p, g x = f x) ∧ ∥g∥ = ∥f∥)
+
+/-- Convert a real number (in particular, a norm) to 𝕜 -/
+noncomputable def coe_from_ℝ (𝕜 : Type*)
+[nondiscrete_normed_field 𝕜] [normed_algebra ℝ 𝕜]
+  (x : ℝ) : 𝕜 :=
+  x • 1
```

Comment:
Not sure if it's worth keeping this as a separate definition

Reply:
The general form of this definition is called `algebra_map ℝ 𝕜`. So you could replace `coe_from_ℝ` with `algebra_map` everywhere, or specialize this definition to:

```lean
/-- Convert a real-valued norm to 𝕜 -/
noncomputable def norm' (𝕜 : Type*) [nondiscrete_normed_field 𝕜] [normed_algebra ℝ 𝕜]
  {E : Type*} [normed_group E] (x : E) : 𝕜 :=
  algebra_map ℝ 𝕜 ∥ x ∥
```

Reply:
Thanks for all your comments; this one in particular helped a lot!